src/test/regress/input/dispatch.source

```diff
@@ -625,3 +625,46 @@ reset gp_print_create_gang_time;
 reset optimizer;
 drop function cleanupAllGangs();
 drop table t_create_gang_time;
+
+--
+-- On QE backends, GUC options are not actually read from the file, but rather
+-- sent to QEs by QD. It means that default settings on segments have source of
+-- PGC_S_CLIENT instead of PGC_S_FILE, which is higher.
+--
+-- GUCs that were changed in the configuration file of the QD should be re-sent
+-- to segments, so segments should have the same options as QD after gpstop -u.
+--
+
+CREATE OR REPLACE FUNCTION show_on_segments(show_name text)
+RETURNS table (guc_name text, guc_setting text)
+EXECUTE ON ALL SEGMENTS AS $$
+BEGIN
+	RETURN QUERY SELECT name, setting FROM pg_settings WHERE show_name = name;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Explicitly set statement_mem through gpconfig, then run gpstop -u. SHOW and
+-- show_on_segments() should return same results.
+
+--start_ignore
+\!gpconfig -c statement_mem -v '125MB'
```

Comment:
What happens and what should happen when we set different values for the coordinator and segments?
```sh
gpconfig -c max_connections -v 100 -m 10
```

Reply:
Here's a comment on how it works:
https://github.com/arenadata/gpdb/blob/45918e2214007e6e9bf6a6b29f867444626486c9/src/backend/cdb/dispatcher/cdbgang.c#L385-L405

In short, on backend instantiation, the coordinator forcefully overwrites GUCs marked with `GUC_GPDB_NEED_SYNC` on QEs.

In case of `gpconfig -c max_connections -v 100 -m 10`, if `max_connections` is marked as `GUC_GPDB_NEED_SYNC`, on QEs it will be the same as on QD. Otherwise, it will not be sent, and it will be out of sync on QD and QEs.